src/Node/Expression/BlockReferenceExpression.php

```diff
@@ -22,7 +22,10 @@
  */
 class BlockReferenceExpression extends AbstractExpression
 {
-    public function __construct(\Twig_NodeInterface $name, ?Node $template, $lineno, $tag = null)
+    /**
+     * @param Node|null $template
+     */
+    public function __construct(\Twig_NodeInterface $name, $template, $lineno, $tag = null)
```

Comment:
The type hint can be added in `2.x` (see the legacy `is_bool()` check below).

Reply:
could even be added in 1.x as the min PHP version has been bumped.